src/identity/configuration/IdentityProviderFactory.ts

```diff
@@ -316,16 +332,30 @@ export class IdentityProviderFactory implements ProviderFactory {
    * Pipes library errors to the provided ErrorHandler and ResponseWriter.
    */
   private configureErrors(config: Configuration): void {
-    config.renderError = async(ctx: KoaContextWithOIDC, out: ErrorOut, error: Error): Promise<void> => {
+    config.renderError = async(ctx: KoaContextWithOIDC, out: ErrorOut, error: errors.OIDCProviderError | Error):
+    Promise<void> => {
       // This allows us to stream directly to the response object, see https://github.com/koajs/koa/issues/944
       ctx.respond = false;
 
-      // OIDC library hides extra details in this field
-      if (out.error_description) {
-        error.message += ` - ${out.error_description}`;
+      // Doesn't really matter which type it is since all relevant fields are optional
+      const oidcError = error as errors.OIDCProviderError;
+
+      // OIDC library hides extra details in these fields
+      if (this.showStackTrace) {
+        if (oidcError.error_description) {
+          error.message += ` - ${oidcError.error_description}`;
+        }
+        if (oidcError.error_detail) {
+          oidcError.message += ` - ${oidcError.error_detail}`;
+        }
+
+        // Also change the error message in the stack trace
+        if (error.stack) {
+          error.stack = error.stack.replace(/.*/u, `${error.name}: ${error.message}`);
+        }
       }
 
-      const result = await this.errorHandler.handleSafe({ error, preferences: { type: { 'text/plain': 1 }}});
+      const result = await this.errorHandler.handleSafe({ error, request: guardStream(ctx.req) });
```

Comment:
The extra `guardStream` surprises me here; do we lose any properties of the request?

Reply:
Our interfaces require guarded streams everywhere. At this point the incoming request has not yet been guarded at some point since it was fully handled by the OIDC library.

Guarding a stream only adds some extra listeners so nothing is lost. In practice it doesn't do anything here since if there was data it has already been read out by the OIDC library. This is just to conform to our typing requirements.